### DISCUSSION TITLE
Topic: should we continue to force errors.Errorf()?

### DISCUSSION THREAD

**Discussion by hallyn** [project-stacker/stacker#376] [General]
The errors package has been "archived" (https://github.com/pkg/errors/issues/245).  errors.Wrapf() is supposed to be replaced by fmt.Errorf("%w").  However, (1) %w output is not liked by some, and more importantly (2) fmt.Errorf does not yet support a way to add a stack trace, which was the original reason for requiring errors.Errorf() in the stacker package.

What should be the official guidance right now?  Should we continue recommending/requiring the errors package, or switch?

**Comment by rchincha** [comment]
errors.Errorf() allows for a generic msg plus custom values.
We should move to a maintained pkg - fmt.Errorf()
There are other ways to dump stack trace if we need that.

**Reply by hallyn** [reply]
Can you make a recommendation for another way to dump stack with an error?

Ideally we would also avoid the problem we've sometimes seen where one error dumps stack several times, by being very clear about "I here am returning a new error, so include my stack" (which might still wrap an error like eperm or enoent), while it's caller says "I'm returning a wrapped error which I expect to include stack" (i.e. fmt.Errorf("%s %w", msg, err)

**Reply by rchincha** [reply]
https://pkg.go.dev/runtime/debug#PrintStack
^ write a fmt.Errorf() wrapper around this?

**Reply by rchincha** [reply]
or this
https://pkg.go.dev/runtime/debug#Stack
